reconcile/skupper_network/integration.py

```diff
@@ -0,0 +1,264 @@
+                cluster=site.cluster.name,
+                namespace=site.namespace.name,
+                resource_type=kind,
+                name=openshift_resource.name,
+                value=openshift_resource,
+            )
+
+
+def fetch_desired_state(
+    ri: ResourceInventory, skupper_sites: Iterable[SkupperSite]
+) -> set[str]:
+    """Fetch desired state of skupper resources in ResourceInventory"""
+    integration_managed_kinds = set()
+    for site in skupper_sites:
+        sc = get_site_controller(site)
+        for resource in sc.resources:
+            openshift_resource = OR(
+                body=resource,
+                integration=QONTRACT_INTEGRATION,
+                integration_version=QONTRACT_INTEGRATION_VERSION,
+            )
+            integration_managed_kinds.add(openshift_resource.kind)
+            # only add desired state if not deleting
+            if not site.delete:
+                ri.initialize_resource_type(
+                    cluster=site.cluster.name,
+                    namespace=site.namespace.name,
+                    resource_type=openshift_resource.kind,
+                )
+                ri.add_desired(
+                    cluster=site.cluster.name,
+                    namespace=site.namespace.name,
+                    resource_type=openshift_resource.kind,
+                    name=openshift_resource.name,
+                    value=openshift_resource,
+                )
+    return integration_managed_kinds
+
+
+def skupper_site_config_changes(ri: ResourceInventory) -> bool:
+    """Check if skupper site config has changes"""
+    changes = False
+    for cluster, ns, resource_type, resource in ri:
+        if resource_type != "ConfigMap":
+            continue
+
+        if (
+            resource["current"]
+            and CONFIG_NAME in resource["current"]
+            and CONFIG_NAME in resource["desired"]
+        ):
+            current = SkupperConfig(**resource["current"][CONFIG_NAME].body["data"])
+            desired = SkupperConfig(**resource["desired"][CONFIG_NAME].body["data"])
+            if current != desired:
+                changes = True
+                logging.error(f"{cluster}/{ns}: Skupper site config has changed")
+    return changes
+
+
+def act(
+    oc_map: OC_Map,
+    ri: ResourceInventory,
+    dry_run: bool,
+    thread_pool_size: int,
+    skupper_sites: Iterable[SkupperSite],
+    integration_managed_kinds: Iterable[str],
+) -> None:
+    """Realize all skupper resources and create skupper site connections."""
+    # skupper-site config map updates are not supported yet
+    # https://github.com/skupperproject/skupper/issues/58
+    # to change the skupper site configuration, the skupper site must be deleted and recreated
+    if skupper_site_config_changes(ri):
+        logging.error(
+            "skupper-site config changes detected. Please delete and recreate the affected skupper site(s)!"
+        )
+        sys.exit(1)
```

Comment:
This means we will not reconcile other networks/sites. Can we just skip the changed sites ?

Reply:
Unfortunately, it's not possible. I don't have the skupper site information anymore; here I'm comparing just configmap in the `ResourceInventory`. Nevertheless, this section should be reached during a PR check only and should not affect the running integration.

Reply:
fair enough!